plugins/biolatency/src/index.tsx

```diff
@@ -0,0 +1,259 @@
+const TRACES_RESOURCE_GROUP = 'gadget.kinvolk.io';
+const IG_TRACE_NAMESPACE = 'gadget';
+const TRACE_BIOLATENCY_RESOURCE_NAME = 'biolatency';
+const ResourceAPI = new K8s.Resource(TRACES_RESOURCE_GROUP, 'v1alpha1', 'traces');
+
+function startLatencyCheck() {
+  return ResourceAPI.patch(
+    [{ op: 'add', path: '/metadata/annotations/gadget.kinvolk.io~1operation', value: 'start' }],
+    IG_TRACE_NAMESPACE,
+    TRACE_BIOLATENCY_RESOURCE_NAME
+  );
+}
+
+function stopLatencyCheck() {
+  return ResourceAPI.patch(
+    [{ op: 'add', path: '/metadata/annotations/gadget.kinvolk.io~1operation', value: 'stop' }],
+    IG_TRACE_NAMESPACE,
+    TRACE_BIOLATENCY_RESOURCE_NAME
+  );
+}
+
+function getLatencyResult(cb: (data: any) => void) {
+  return ResourceAPI.apiEndpoint.get(
+    IG_TRACE_NAMESPACE,
+    TRACE_BIOLATENCY_RESOURCE_NAME,
+    function (data: any) {
+      cb(data);
+    }
+  );
+}
+
+const LATENCY_CHECK_NOT_STARTED = -1;
+const LATENCY_CHECK_STOPPED = 0;
+const LATENCY_CHECK_STARTED = 1;
+
+function CustomComponent(props: any) {
+  const { resource } = props;
+  const [outputString, setOutputString] = React.useState('');
+  const [outputError, setOutputError] = React.useState(null);
+  const [latencyCheckStatus, setLatencyCheckStatus] = React.useState(LATENCY_CHECK_NOT_STARTED);
+  let componentToRender = (
+    <LatencyCheckInitial
+      setLatencyCheckStatus={() => setLatencyCheckStatus(LATENCY_CHECK_STARTED)}
+    />
+  );
+  if (resource && resource.kind === 'Node') {
+    switch (latencyCheckStatus) {
+      case 0:
+        componentToRender = (
+          <LatencyCharts
+            outputString={outputString}
+            setLatencyCheckStatus={() => setLatencyCheckStatus(LATENCY_CHECK_STARTED)}
+            outputError={outputError}
+          />
+        );
+        break;
+      case 1:
+        componentToRender = (
+          <TimerComponent
+            setLatencyCheckStatus={setLatencyCheckStatus}
+            setOutputString={setOutputString}
+            setOutputError={setOutputError}
+          />
+        );
+        break;
+      default:
+        componentToRender = (
+          <LatencyCheckInitial
+            setLatencyCheckStatus={() => setLatencyCheckStatus(LATENCY_CHECK_STARTED)}
+          />
+        );
+    }
+  }
+  return <Box>{componentToRender}</Box>;
+}
+
+function LatencyCharts(props: any) {
+  const { outputString, setLatencyCheckStatus, outputError } = props;
+  const [chartData, setChartData] = React.useState(null);
+
+  function prepareChartData(outputString: string) {
+    const chartData = outputString.split('\n');
+    const data = [];
+    for (let i = 3; i < chartData.length; i++) {
+      const dataChild = chartData[i].split(/[:|/|]/);
+      if (dataChild[0] && dataChild[1] && dataChild[2]) {
+        const usecs = dataChild[0].trim().replace('->', '-');
+        data.push({
+          usecs: usecs,
+          count: parseInt(dataChild[1].trim()),
+        });
+      }
+    }
```

Comment:
Some comments on what format this expects and what this does would make the code easier to maintain when we have to touch it in a few months.